core/src/main/java/com/zextras/carbonio/files/message_broker/MessageBrokerManagerImpl.java

```diff
@@ -0,0 +1,68 @@
+// SPDX-FileCopyrightText: 2024 Zextras <https://www.zextras.com>
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package com.zextras.carbonio.files.message_broker;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.zextras.carbonio.files.config.FilesConfig;
+import com.zextras.carbonio.files.dal.repositories.interfaces.NodeRepository;
+import com.zextras.carbonio.files.message_broker.consumers.UserStatusChangedConsumer;
+import com.zextras.carbonio.files.message_broker.interfaces.MessageBrokerManager;
+import com.zextras.carbonio.message_broker.MessageBrokerClient;
+import com.zextras.carbonio.message_broker.config.enums.Service;
+import com.zextras.carbonio.message_broker.events.generic.BaseEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is basically a repository for MessageBroker (RabbitMQ).
+ * This class has the responsibility of starting consumers of events that need to be listened to.
+ */
+@Singleton
+public class MessageBrokerManagerImpl implements MessageBrokerManager {
+
+  private static final Logger logger = LoggerFactory.getLogger(MessageBrokerManagerImpl.class);
+
+  private final MessageBrokerClient messageBrokerClient;
+  private final NodeRepository nodeRepository;
+
+  @Inject
+  public MessageBrokerManagerImpl(
+      FilesConfig filesConfig,
```

Comment:
Let the `MessageBrokerClient` be provided by a GuiceProvider and injected directly in the constructor. You are free to create a new Provider class or use the `@Provider` annotation in the `FilesModule` class